src/Health/HealthController.ts

```diff
@@ -0,0 +1,205 @@
+import BitcoinCore = require('bitcoin-core')
+import { inject, injectable } from 'inversify'
+import { Collection, Db } from 'mongodb'
+import * as Pino from 'pino'
+
+import { childWithFileName } from 'Helpers/Logging'
+import { IPFSHashFailure } from 'Interfaces'
+import { IPFS } from './IPFS'
+
+@injectable()
+export class HealthController {
+  private readonly db: Db
+  private readonly collection: Collection
+  private readonly bitcoinCore: BitcoinCore
+  private readonly logger: Pino.Logger
+  private readonly ipfs: IPFS
+
+  constructor(
+    @inject('Logger') logger: Pino.Logger,
+    @inject('DB') db: Db,
+    @inject('BitcoinCore') bitcoinCore: BitcoinCore,
+    @inject('IPFS') ipfs: IPFS
+  ) {
+    this.logger = childWithFileName(logger, __filename)
+    this.db = db
+    this.collection = this.db.collection('health')
+    this.bitcoinCore = bitcoinCore
+    this.ipfs = ipfs
+  }
+
+  async checkMongoConnection(): Promise<any> {
+    this.logger.trace('MongoHealth', 'Checking Mongo...')
+    const connection = await this.db.stats()
+    const mongoIsConnected = connection.ok === 1 ? true : false
```

Comment:
You can just do `const mongoIsConnected = connection.ok === 1` here, leaving ` ? true : false` out has the same effect